internal/handlers/common/update.go

```diff
@@ -118,5 +122,77 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 		}
 	}
 
-	return true, nil
+	return changed, nil
+}
+
+// ValidateUpdateOperators validates update statement.
+func ValidateUpdateOperators(update *types.Document) (err error) {
+	set, err := getUpdateOperatorDocument("$set", update)
+	if err != nil {
+		return err
+	}
+	_, err = getUpdateOperatorDocument("$setOnInsert", update)
+	if err != nil {
+		return err
+	}
+
+	inc, err := getUpdateOperatorDocument("$inc", update)
+	if err != nil {
+		return err
+	}
+
+	if set != nil && inc != nil {
```

Comment:
How that would look like with more operators?

Reply:
> How that would look like with more operators?

1. what is your suggestion?
2. can it be solved when more operators appear?

Reply:
If we know how to scale that code for more update operators, I think we should do that now. If we don't know, well, then maybe we should write this code some other way? We know for sure that more operators will be added